core/src/main/java/org/apache/calcite/avatica/MetaImpl.java

```diff
@@ -139,15 +134,8 @@ protected Getter createGetter(int ordinal) {
       return list;
     case RECORD:
     case RECORD_PROJECTION:
```

Comment:
With the current PR the behavior of `RECORD` and `RECORD_PROJECTION` seems to be exactly the same. Do we need to keep both?

Reply:
I agree, we can unify them (LGTM in your branch)